### PR TITLE
12.0 odoo_backup_sh: fix

### PR DESCRIPTION
**Pull request by em230418**

### CODE REVIEW

**Comment by em230418** [comment]
Missing new entry in changelog, adding

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class BackupConfig(models.Model):
     _description = 'Backup Configurations'
     _rec_name = 'database'
 
-    DATABASE_NAMES = [(db, db) for db in odoo.service.db.list_dbs() if db != 'session_store']
+    DATABASE_NAMES = [(db, db) for db in odoo.service.db.list_dbs(force=True) if db != 'session_store']
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: Actually, I'm not sure about this solution. Imagine we have different companies that share the same odoo instance. Those companies are allowed to use backup module. Then any admin can get backup of all companies.

What we can do istead is use current database only when list_db is disabled

File: odoo_backup_sh/__manifest__.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "category": "Backup",
     # "live_test_url": "",
     "images": ['images/backupsh.jpg'],
-    "version": "12.0.1.0.0",
+    "version": "12.0.1.0.1",
```
Comment (yelizariev) [inline odoo_backup_sh/__manifest__.py]: No need to chage version as the module is not published

Comment (em230418) [reply odoo_backup_sh/__manifest__.py]: Done

File: odoo_backup_sh/models/odoo_backup_sh.py (hunk 1 above)
Comment (em230418) [reply odoo_backup_sh/models/odoo_backup_sh.py]: > What we can do i[n]stead is use current database only when list_db is disabled

Doing like this then: https://github.com/odoo/odoo/blob/8f4b98e67fe49516c0c830524de351db67c58357/odoo/service/db.py#L369

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 3]
```diff
@@ -41,7 +41,12 @@ class BackupConfig(models.Model):
     _description = 'Backup Configurations'
     _rec_name = 'database'
 
-    DATABASE_NAMES = [(db, db) for db in odoo.service.db.list_dbs() if db != 'session_store']
+    if config['list_db']:
+        DATABASE_NAMES = [(db, db) for db in odoo.service.db.list_dbs() if db != 'session_store']
+    elif config['db_name']:
+        DATABASE_NAMES = [(db, db) for db in odoo.tools.config['db_name'].split(',') if db != 'session_store']
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: use ``self.env.cr`` instead. ``db_name`` may be not set or has another value

Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: exception for session_store can be deleted as it's from abandoned docker deployment

Comment (em230418) [reply odoo_backup_sh/models/odoo_backup_sh.py]: `self` is not available in this context. Also, I do not understand, which cursor from which database can be used to define field `database` field in line 53. The only database connection, that is applicable (for me at least) is https://github.com/odoo/odoo/blob/8f4b98e67fe49516c0c830524de351db67c58357/odoo/service/db.py#L374 , but then I would be better just to use rejected `force=True` solution.

Comment (em230418) [reply odoo_backup_sh/models/odoo_backup_sh.py]: Using odooup I still have `list_dbs()` array with 'session_store' value.

Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: ok

Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: Then you need to use request.env.cr

Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: You need to use current database. I've explained why force=True is not an option

Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: I said wrong about  ``request.env.cr``. You need to make a method to compute list of available options.

 database = fields.Selection(selection=_selection_database)

Comment (em230418) [reply odoo_backup_sh/models/odoo_backup_sh.py]: Could not find another way, where we can get databases. Checked odoo code. 
https://github.com/odoo/odoo/blob/cb2862ad2a60ff4ce66c14e7af2548fdf6fc5961/odoo/cli/server.py#L135-L152
If no db_name set, odoo works without any databases.

Also could not find code, where db_name is modified.

So, I insist in using current solution.

Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: request.env.cr.dbname (or db_name) is always available

Comment (em230418) [reply odoo_backup_sh/models/odoo_backup_sh.py]: request variable (`from odoo.http import request`) is usually evaluated inside http controller. In this context, evaluating request will `RuntimeError: object unbound`

Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: I know. I told that few messages before: https://github.com/it-projects-llc/addons-dev/pull/688#discussion_r309234706

Comment (yelizariev) [reply odoo_backup_sh/models/odoo_backup_sh.py]: https://github.com/odoo/odoo/blob/12.0/addons/mail/wizard/mail_compose_message.py#L100-L105

Comment (em230418) [reply odoo_backup_sh/models/odoo_backup_sh.py]: Fixed using https://github.com/it-projects-llc/addons-dev/pull/688/commits/64603c1c9cf1a7f46c329a8def5a44843767c1cc

**Comment by yelizariev** [comment]
I approve to merge it now

**Pull request merged**